src/gen9_avc_encoder.c

```diff
@@ -5634,15 +5913,1028 @@ gen9_avc_kernel_sfd(VADriverContextP ctx,
+    }
+
+    /* BRC distortion data buffer for I frame */
+
+    if (!mbenc_i_frame_dist_in_use) {
+        if (avc_state->mb_disable_skip_map_enable) {
+            gpe_resource = &(avc_ctx->res_mb_disable_skip_map_surface);
+            gen9_add_buffer_2d_gpe_surface(ctx, gpe_context,
+                                           gpe_resource,
+                                           1,
+                                           I965_SURFACEFORMAT_R8_UNORM,
+                                           GEN9_AVC_MBENC_FORCE_NONSKIP_MB_MAP_INDEX);
+        }
+
+        if (avc_state->sfd_enable) {
+            size = 128 / sizeof(unsigned long);
+            gpe_resource = &(avc_ctx->res_sfd_output_buffer);
+            gen9_add_buffer_gpe_surface(ctx,
+                                        gpe_context,
+                                        gpe_resource,
+                                        0,
+                                        size / 4,
+                                        0,
+                                        GEN8_AVC_MBENC_STATIC_FRAME_DETECTION_OUTPUT_CM);
+        }
+    }
+    return;
+}
+
+static void
+gen8_avc_set_curbe_scaling4x(VADriverContextP ctx,
+                             struct encode_state *encode_state,
+                             struct i965_gpe_context *gpe_context,
+                             struct intel_encoder_context *encoder_context,
+                             void *param)
+{
+    gen8_avc_scaling4x_curbe_data *curbe_cmd;
+    struct scaling_param *surface_param = (struct scaling_param *)param;
+
+    curbe_cmd = i965_gpe_context_map_curbe(gpe_context);
+
+    if (!curbe_cmd)
+        return;
+
+    memset(curbe_cmd, 0, sizeof(gen9_avc_scaling4x_curbe_data));
+
+    curbe_cmd->dw0.input_picture_width  = surface_param->input_frame_width;
+    curbe_cmd->dw0.input_picture_height = surface_param->input_frame_height;
+
+    curbe_cmd->dw1.input_y_bti = GEN8_SCALING_FRAME_SRC_Y_CM;
+    curbe_cmd->dw2.output_y_bti = GEN8_SCALING_FRAME_DST_Y_CM;
+
+
+    curbe_cmd->dw5.flatness_threshold = 0;
+    if (surface_param->enable_mb_flatness_check) {
+        curbe_cmd->dw5.flatness_threshold = 128;
+        curbe_cmd->dw8.flatness_output_bti_top_field = 4;
+    }
+
+    curbe_cmd->dw6.enable_mb_flatness_check = surface_param->enable_mb_flatness_check;
+    curbe_cmd->dw6.enable_mb_variance_output = surface_param->enable_mb_variance_output;
+    curbe_cmd->dw6.enable_mb_pixel_average_output = surface_param->enable_mb_pixel_average_output;
+
+    if (curbe_cmd->dw6.enable_mb_variance_output ||
+        curbe_cmd->dw6.enable_mb_pixel_average_output) {
+        curbe_cmd->dw10.mbv_proc_states_bti_top_field  = GEN8_SCALING_FIELD_TOP_MBVPROCSTATS_DST_CM;
+        curbe_cmd->dw11.mbv_proc_states_bti_bottom_field = GEN8_SCALING_FIELD_BOT_MBVPROCSTATS_DST_CM;
+    }
+
+    i965_gpe_context_unmap_curbe(gpe_context);
+    return;
+}
+
+static void
+gen8_avc_send_surface_scaling(VADriverContextP ctx,
```

Comment:
it is almost the same as gen9_avc_send_surface_scaling. is it possible to reuse it?